raster_analysis/tiling.py

```diff
@@ -1,144 +1,143 @@
-
 
 @xray_recorder.capture("Merge Tiled Geometry Results")
 def merge_tile_results(tile_results, groupby_columns):
-    concatted_tile_results = concat_tile_results(tile_results)
-
-    groupby_results = np.array([concatted_tile_results[col] for col in groupby_columns])
-    column_maxes = [col.max() + 1 for col in groupby_results]
-    linear_indices = np.ravel_multi_index(groupby_results, column_maxes).astype(
-        np.uint32
-    )
-
-    unique_values, inv = np.unique(linear_indices, return_inverse=True)
-    unique_value_combinations = np.unravel_index(unique_values, column_maxes)
-
-    merged_tile_results = dict(zip(groupby_columns, unique_value_combinations))
-
-    for col_name, col_data in concatted_tile_results.items():
-        if col_name not in groupby_columns:
-            arr = np.array(col_data)
-            merged_tile_results[col_name] = np.bincount(inv, weights=col_data).astype(
-                arr.dtype
+    if not groupby_columns:
+        dataframes = [pd.DataFrame(result, index=[0]) for result in tile_results]
+        merged_df: pd.DataFrame = pd.concat(dataframes)
+        return merged_df.sum().to_dict()
+
+    dataframes = [pd.DataFrame(result) for result in tile_results]
+    merged_df: pd.DataFrame = pd.concat(dataframes)
+
+    grouped_df: pd.DataFrame = merged_df.groupby(groupby_columns).sum()
+    result_df: pd.DataFrame = grouped_df.sort_values(groupby_columns).reset_index()
+
+    # convert ordinal dates to readable dates
+    for col in groupby_columns:
+        if "__date" in col:
+            result_df[col] = result_df[col].apply(
+                lambda val: date.fromordinal(val).strftime("%Y-%m-%d")
+            )
+        elif "__isoweek" in col:
+            result_df[col.replace("__isoweek", "__year")] = result_df[col].apply(
+                lambda val: date.fromordinal(val).isocalendar()[0]
+            )
+            result_df[col] = result_df[col].apply(
+                lambda val: date.fromordinal(val).isocalendar()[1]
             )
 
-    for col, arr in merged_tile_results.items():
-        merged_tile_results[col] = arr.tolist()
-
-    return merged_tile_results
+    return result_df.to_dict(orient="records")
 
 
-def concat_tile_results(tile_results):
-    result = deepcopy(tile_results[0])
+def process_tiled_geoms(tiles, geoprocessing_params, request_id, fanout_num):
+    geom_count = len(tiles)
+    geoprocessing_params["write_to_dynamo"] = True
+    geoprocessing_params["analysis_id"] = request_id
+    logger.info(f"Processing {geom_count} tiles")
 
-    for col in result.keys():
-        for table in tile_results[1:]:
-            result[col] += table[col]
+    tile_geojsons = [mapping(tile) for tile in tiles]
+    tile_chunks = [
+        tile_geojsons[x : x + fanout_num]
+        for x in range(0, len(tile_geojsons), fanout_num)
+    ]
+    fanout_lambda = os.environ["FANOUT_LAMBDA"]
 
-    return result
+    from raster_analysis.boto import invoke_lambda
```

Comment:
why import here in line, not at the top of module?

Reply:
this was kind of hack so I could monkey patch it in the test code. I'll see if there's a better way